client/extensions/woocommerce/state/sites/settings/stripe-connect-account/test/handlers.js

```diff
@@ -106,6 +110,77 @@ describe( 'handlers', () => {
 		} );
 	} );
 
+	describe( '#handleAccountDisconnect', () => {
+		test( 'should dispatch a POST request', () => {
+			const siteId = '123';
+			const dispatch = spy();
+			const action = disconnectAccount( siteId );
+			handleAccountDisconnect( { dispatch }, action );
+			expect( dispatch ).to.have.been.calledWithMatch( {
+				type: WPCOM_HTTP_REQUEST,
+				body: {
+					path: '/wc/v1/connect/stripe/account/deauthorize&_method=POST',
+					body: JSON.stringify( { deauthorize: true } ),
+				},
+				method: 'POST',
+				path: `/jetpack-blogs/${ siteId }/rest-api/`,
+				query: {
+					json: true,
+					apiVersion: '1.1',
+				},
+			} );
+		} );
+	} );
+
+	describe( '#handleAccountDisconnectSuccess()', () => {
+		test( 'should dispatch update on success', () => {
+			const siteId = '123';
+			const store = {
+				dispatch: spy(),
+			};
+			const response = {
+				data: {
+					success: true,
+				},
+			};
+
+			const action = disconnectAccount( siteId );
+			handleAccountDisconnectSuccess( store, action, response );
+			expect( store.dispatch ).calledWith( {
+				type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_DISCONNECT_COMPLETE,
+				siteId,
+			} );
+		} );
+	} );
+
+	describe( '#handleAccountDisconnectFailure()', () => {
+		test( 'should dispatch update with error', () => {
+			const siteId = '123';
+			const store = {
+				dispatch: spy(),
+			};
+			const response = {
+				data: {
+					body: {
+						data: {
+							message: '.',
+						},
+						success: false,
+					},
+					status: 400,
+				},
+			};
+
+			const action = disconnectAccount( siteId );
+			handleAccountDisconnectFailure( store, action, response.data.body.data.message );
```

Comment:
Minor: It looks like this is the structure that would be used if you were making a call with http://v2.wp-api.org/reference/global.html#envelope -- While this is still a valid test for the behavior, you could probably reduce the example here a bit and just pass a string.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/19244/commits/2d81a7a15a8b02918fd6b2ef39b9a54500e0b52c (big rewrite of tests)